src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -398,13 +415,30 @@ public void DefManyType()
                 AllowsSideEffects = false,
             };
 #pragma warning disable CS0618 // Type or member is obsolete
-            recalcEngine.DefineType("A = Type({ num: Number}); B = Type({ a: A}); C = Type({ b: B, a: A});", parserOptions);
+            Assert.False(recalcEngine.DefineType("A = Type({ num: Number}); B = Type({ a: A}); C = Type({ b: B, a: A});", parserOptions).Any());
 #pragma warning restore CS0618 // Type or member is obsolete
-            IEnumerable<ExpressionError> enumberable = recalcEngine.DefineFunctions("foo(p: C): Number = p.b.a.num + p.a.num;").Errors;
-            Assert.False(enumberable.Any());
+            IEnumerable<ExpressionError> enumerable = recalcEngine.DefineFunctions("foo(p: C): Number = p.b.a.num + p.a.num;").Errors;
+            Assert.False(enumerable.Any());
             Assert.Equal(1.5, recalcEngine.Eval("foo({b: {a: {num: Float(0.5)}}, a: {num: Float(1.0)}})").ToObject());
         }
 
+        [Fact]
+        public void TypeAlias()
+        {
+            var config = new PowerFxConfig();
+            var recalcEngine = new RecalcEngine(config);
+            var parserOptions = new ParserOptions()
+            {
+                AllowsSideEffects = false,
+            };
+#pragma warning disable CS0618 // Type or member is obsolete
+            Assert.False(recalcEngine.DefineType("Weight = Type(Number);", parserOptions).Any());

```

Comment:
>Assert.False(recalcEngine.DefineType("Weight = Type(Number);", parserOptions).Any()); [](http://example.com/codeflow?start=12&length=85)

Why do we wanna use obsolete?
Can we do, Assert.Empty()?

Reply:
We wanna use obsolete because this is still preview functionality, and we wanna end up replacing these disperate entry points with a single entrypoint for udfs and udts, and possibly also for named formulas. and yes to the second thing